net: socket: udp: net.socket.udp.port_range reports FLASH overflow

**Describe the bug**
The test `.tests/net/socket/udp/net.socket.udp.port_range` fails.

Observed for 

- `nrf5340dk/nrf5340/cpuapp/ns`,
- `nrf9160dk@0.14.0/nrf9160/ns`.

**To Reproduce**
Steps to reproduce the behavior:
1. have `nrf5340dk/nrf5340` connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/net/socket/udp/ -p nrf5340dk/nrf5340/cpuapp/ns --device-testing --device-serial /dev/ttyACM1 -v --inline-logs --west-flash="--erase"`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre1.elf section `datas' will not fit in region `FLASH'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: region `FLASH' overflowed by 4696 bytes
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
- Toolchain Zephyr SDK 0.16.8
- Commit SHA or Version used: `v4.0.0-735-geb4f2fb44a0e`

**Additional information:**
Introduced by commit 5531692bcc7e8cdab19b3ccbf72a2dd2bc361e1b

--- a/tests/net/socket/udp/testcase.yaml
+++ b/tests/net/socket/udp/testcase.yaml
@@ -5,6 +5,7 @@ common:
     - socket
     - udp
   min_ram: 21
+  min_flash: 194
   filter: CONFIG_FULL_LIBC_SUPPORTED
 tests:
   net.socket.udp:

